ipywebrtc/webrtc.py

```diff
@@ -71,93 +71,168 @@ def _valid_fps(self, proposal):
+        """Create a `ImageStream` from a url.
+        This will create a `ImageStream` from an Image using its url
+
+        Parameters
+        ----------
+        url: str
+            The url of the file that will be used for the .image trait.
+        **kwargs:
+            Extra keyword arguments for `ImageStream`
+        Returns an `ImageStream`.
+        """
+        return cls(image=Image.from_url(url), **kwargs)
 
 
 @register
 class VideoStream(MediaStream):
-    """Represents a media source by a video.
-
-    The `value` of this widget accepts a byte string. The byte string is the
-    raw video data that you want the browser to display.  You can explicitly
-    define the format of the byte string using the `format` trait (which
-    defaults to 'mp4').
-    If you pass `"url"` to the `"format"` trait, `value` will be interpreted
-    as a URL as bytes encoded in ascii.
-    """
+    """Represent a stream of a video element"""
     _model_name = Unicode('VideoStreamModel').tag(sync=True)
 
-    format = Unicode('mp4', help="The format of the video.").tag(sync=True)
-    value = Bytes(None, allow_none=True, help="The video data as a byte string.").tag(sync=True)
-    play = Bool(True, help='Plays the video or pauses it.').tag(sync=True)
-    loop = Bool(True, help='When true, the video will start from the beginning after finishing.').tag(sync=True)
+    video = Instance(
+        Video,
+        allow_none=False,
+        help="An ipywidgets.Video instance that will be the source of the media stream."
+    ).tag(sync=True, **widget_serialization)
+    play = Bool(True, help='Plays the videostream or pauses it.').tag(sync=True)
 
     @classmethod
-    def from_file(cls, f, **kwargs):
-        """Create a `VideoStream` from a local file or file object.
+    def from_file(cls, filename, **kwargs):
+        """Create a `VideoStream` from a local file.
 
         Parameters
         ----------
-        f: str or file
-            The path or file object that will be read and its bytes assigned
-            to the value trait.
+        filename: str
+            The location of a file to read into the value from disk.
         **kwargs:
             Extra keyword arguments for `VideoStream`
-        Returns an `VideoStream` with the value set from the content of a file.
+        Returns an `VideoStream`.
         """
-        if isinstance(f, string_types):
-            with open(f, 'rb') as f:
-                return cls(value=f.read(), **kwargs)
-        else:
-            if 'format' not in kwargs:
-                ext = os.path.splitext(f)[1]
-                if ext:
-                    kwargs['format'] = ext[1:]  # remove the .
-            return cls(value=f.read(), **kwargs)
+        autoplay = True
+        if kwargs.get('play') is not None:
+            autoplay = kwargs.get('play')
+
+        video = Video.from_file(
+            filename,
+            autoplay=autoplay,
+            controls=False
+        )
+
```

Comment:
I'm not a big fan of this formatting, I think it fits perfectly on one line, versus 5.

Reply:
will do :)